moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -127,36 +127,54 @@ def add_plane(self, name, pose, normal=(0, 0, 1), offset=0):
         co = self.make_plane(name, pose, normal, offset)
         self.__submit(co, attach=False)
 
-    def attach_object(self, attached_collision_object):
-        """Attach an object in the planning scene"""
-        self.__submit(attached_collision_object, attach=True)
+    def attach_object(self, object, link, touch_links=None):
+        """Attach an object to the given link"""
+        if isinstance(object, str):
+            object = self.__make_existing(object)
+        if isinstance(object, CollisionObject):
+            object = AttachedCollisionObject(object=object)
+
+        object.link_name = link
```

Comment:
```suggestion
     def attach_object(self, object, link=None, touch_links=None):
        """Attach an object to the given link"""
        if isinstance(object, str):
            object = self.__make_existing(object)
        if isinstance(object, CollisionObject):
            object = AttachedCollisionObject(object=object)

    if link is not None:
        object.link_name = link
```

Looks like github does not support scattered diffs in the suggestions -.-
Anyway, I propose to support passing an `AttachedCollisionObject` which already specifies the link in the message.